agreement/pseudonode_test.go

```diff
@@ -222,7 +222,12 @@ func TestPseudonode(t *testing.T) {
 		}
 		messageEvent, typeOk := ev.(messageEvent)
 		assert.True(t, true, typeOk)
+		// Verify votes are recorded - everyone is voting and proposing blocks.
+		keyManager.mutex.Lock()
+		assert.Equal(t, startRound, keyManager.recording[messageEvent.Input.Vote.R.Sender][account.Vote])
+		assert.Equal(t, startRound, keyManager.recording[messageEvent.Input.Vote.R.Sender][account.BlockProposal])
 		events[messageEvent.t()] = append(events[messageEvent.t()], messageEvent)
+		keyManager.mutex.Unlock()
```

Comment:
Could you create a key manager method 
```golang
ValidateVoteRound(t, voteRound, act, account.ParticipationAction)
```
that would take the lock, conduct the test and return ?
i.e. given that this key manager is a testing key manager, it's perfectly fine to add testing method to it.